File: ISS Emergency/Assets/Scripts/UniqueIDManager.js
#pragma strict

// Hannes Helmholz
//
// 


@script ExecuteInEditMode()

private class UniqueIDManager extends MonoBehaviour {
	
	// =============================================================================
	// MEMBERS ---------------------------------------------------------------------
	public var UniqueIDs:UniqueID[];
	// =============================================================================
	
	
	
	// =============================================================================
	// METHODS UNITY ---------------------------------------------------------------
	
	function Start() {
		UniqueIDs = GameObject.FindObjectsOfType(UniqueID) as UniqueID[];
		for (var i:int = 0; i < UniqueIDs.length; i++)
			UniqueIDs[i].ID = i;
	}
	// =============================================================================
	
	
	
	// =============================================================================
	// METHODS ---------------------------------------------------------------------
	
	public function GetGameObjectByID(id:int):GameObject {
		return UniqueIDs[id].gameObject;
	}
	// =============================================================================
}

File: Assets/Scripts/UniqueIDManager.cs

// Hannes Helmholz
//
// 

using UnityEngine;

[ExecuteInEditMode]

/** Klasse, welche UniqueID's an GameObjekte
 *  vergibt und somit eindeutig macht. Außerdem
 *  können GameObjekte anhand ihrer UniqueID
 *  ermittelt werden.
 */
public class UniqueIDManager : MonoBehaviour
{
 
    // =============================================================================
    // MEMBERS ---------------------------------------------------------------------
    public UniqueID[] UniqueIDs;
    // =============================================================================
 
 
 
    // =============================================================================
    // METHODS UNITY ---------------------------------------------------------------
 
    void Start ()
    {
        UniqueIDs = FindObjectsOfType ( typeof(UniqueID) ) as UniqueID[];
        for ( int i = 0; i < UniqueIDs.Length; i++ )
        {
            UniqueIDs[i].ID = i;
        }
    }
    // =============================================================================
 
 
 
    // =============================================================================
    // METHODS ---------------------------------------------------------------------
 
    public GameObject GetGameObjectByID ( int id )
    {
        return UniqueIDs[id].gameObject;
    }
    // =============================================================================
}